docs/developers/style-guide.md

```diff
@@ -2,6 +2,94 @@
 title: Style Guide
 ---
 
-!!! warning
+# Style Guide
 
-    This section is work in progress. Please check back soon for the updated content.
+This section collects general coding style guidelines used in Fast-LLM.
+Following these will ensure a swift reviewing process and will help maintaining consistency and readability.
+Note that while we try to enforce these principles,
+exceptions may be allowed on a case-by-case basis, for example if they noticeably improve readability.
+
+As a general principle, **Fast-LLM prioritize code readability and maintainability** over conciseness,
+coding speed or individual programmer's preferences.
+Most of the style choices below are based on this principle.
+
+## 🎯 Basic Style
+
+Unless otherwise specified, Fast-LLM follows the [PEP 8](https://peps.python.org/pep-0008/) coding style.
+This style (and many other conventions) is enforced with automatic formatting through a [pre-commit](https://pre-commit.com/) git hook.
+
+Make sure these git hooks are installed by running
+```bash
+pip install pre-commit
+pre-commit install
+```
+
+!!! note "More on automated formatting"
+
+    Fast-LLM's automated formatting includes [Black](https://black.readthedocs.io/en/stable/),
+    [isort](https://pycqa.github.io/isort/), [autoflake](https://github.com/PyCQA/autoflake), and a few other packages.
+    See Fast-LLM's [pre-commit configuration](https://github.com/ServiceNow/Fast-LLM/blob/main/.pre-commit-config.yaml) for more details.
+
+## 📚 Naming Conventions
+
+Names should be as self-explanatory as possible, within reason.
+This includes python identifiers (classes, variables, methods, modules, etc.), file names and configuration parameters.
+For example:
+
+* Avoid abbreviations, especially domain-specific ones. Ex. `bs -> batch_size`.
+* Avoid meaningless variable names such as `x`, except for generic methods.
+* Avoid redundancies especially for configuration parameters, ex. `data.data_type` -> `data.type`.
+* Avoid name parts that refer to the data type, ex. `num`. Use type hints instead.
+
+Note that these conventions are enforced more strictly on user-facing names since they are more difficult to change,
+for example configuration parameters and the public interface of core classes and modules.
+
+## 🛬 Imports
+
+We use the following conventions for imports (other than those enforced by isort):
+
+* Import standard library and third party modules by module (ex. `import package.module`, not `from package.module import method`).
```

Comment:
Why prefer `import package.module` universally? Fast-LLM is the only library I know that does that. I think this needs to be well motivated.

Reply:
This is a bit of a personal preference, but Fast-LLM doesn't use that many external libraries and nearly all of them keep module identifiers short so it's not a big difference. Main benefits are:
* Consistency: we're not doing things different for every file depending on who wrote it.
* We know what package an identifier comes from with just a quick look at the code, no digging at all.
* It's especially useful for commonly used names that could come from multiple packages, or that are also present in Fast-LLM (one example: Fast-LLM's config has many names in common with dataclasses, omegaconf and pydantic)
* We could allow exceptions, but this would be subjective and hard to manage.

Reply:
Thanks for spelling it out. Could you add points 1-3 to the doc, please?